src/elm/Page/Dashboard/Analysis.elm

```diff
@@ -130,31 +144,97 @@ view ({ shared } as loggedIn) model =
             text ""
 
 
-viewAutoCompleteProfile : LoggedIn.Model -> Model -> Maybe Profile -> Html Msg
-viewAutoCompleteProfile { shared } model maybeProfile =
-    case model.communityStatus of
-        LoadedCommunity community ->
-            let
-                selectedUsers =
-                    Maybe.map (\v -> [ v ]) maybeProfile
-                        |> Maybe.withDefault []
-            in
-            div []
-                [ Html.map SelectMsg
-                    (Select.view
-                        (selectConfiguration shared False)
-                        model.autoCompleteState
-                        community.members
-                        selectedUsers
+viewFilters : LoggedIn.Model -> Model -> Html Msg
+viewFilters ({ shared } as loggedIn) model =
+    let
+        t : String -> String
+        t =
+            I18Next.t shared.translations
+
+        text_ s =
+            text (t s)
+    in
+    div [ class "mt-4 mb-12" ]
+        [ div []
+            [ span [ class "input-label" ]
+                [ text_ "all_analysis.filter.user" ]
+            , case model.communityStatus of
+                LoadedCommunity community ->
+                    let
+                        selectedUsers =
+                            Maybe.map (\v -> [ v ]) model.filters.profile
+                                |> Maybe.withDefault []
+                    in
+                    div []
+                        [ Html.map SelectMsg
+                            (Select.view
+                                (selectConfiguration shared False)
+                                model.autoCompleteState
+                                community.members
+                                selectedUsers
+                            )
+                        , viewSelectedVerifiers loggedIn selectedUsers
+                        ]
+
+                _ ->
+                    text ""
+            ]
+        , div [ class "mt-6" ]
+            [ span [ class "input-label" ] [ text_ "all_analysis.filter.status.label" ]
+            , select
+                [ class "input w-full mb-2 border form-select border-gray-500 rounded-sm"
+                , Html.Events.on "change"
+                    (Decode.map
+                        (\val ->
+                            case val of
+                                "approved" ->
+                                    SelectStatusFilter Approved
+
+                                "rejected" ->
+                                    SelectStatusFilter Rejected
+
+                                "pending" ->
+                                    SelectStatusFilter Pending
+
+                                _ ->
+                                    SelectStatusFilter All
+                        )
+                        Html.Events.targetValue
```

Comment:
Would this be better as a named function? That way we can reuse/change the decoding in the future without going into view code

Reply:
probably we won't. See, `SelectStatusFilter` is used only for mapping the options in to the page type. That is, unless we also bring the type to an external place, it will be more difficult to be reusable.

In general I prefer to not optimize prematurely, but rather optimize once we see window for it clearly